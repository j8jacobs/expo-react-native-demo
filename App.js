import { StatusBar } from "expo-status-bar";
import { StyleSheet, Text, View } from "react-native";
import { SafeAreaProvider } from "react-native-safe-area-context";
import Constants from "expo-constants";

export default function App() {
  return (
    <SafeAreaProvider style={styles.frameContainer}>
      <StatusBar style="auto" />
      <View style={styles.container}>
        <View style={styles.card}>
          <Text>Hello</Text>
        </View>
      </View>
    </SafeAreaProvider>
  );
}

const styles = StyleSheet.create({
  frameContainer: {
    flex: 1,
    marginTop: Constants.statusBarHeight,
  },
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    paddingTop: 20,
  },
  card: {
    width: "90%",
    height: 200,
    borderWidth: 1,
    margin: 12,
    padding: 12,
    borderRadius: 8,
  },
});
