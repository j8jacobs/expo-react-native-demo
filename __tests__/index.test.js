import React from "react";
import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react-native";

import App from "../app/index";

describe("<App />", () => {
  // basic testing functionalities: basic render and snapshot
  it("renders", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree.children.length).toBe(1);
  });
  it("renders correctly", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  // more complex testing (experiments)
  it("All tabs are there", () => {
    render(<App />);
    // getByText must match exact text
    // if it's not found, it will throw and the test will fail
    screen.getByText("Purpose");
    screen.getByText("Navigation");
    screen.getByText("Kanye West Quote Generator");
    screen.getByText("Build a Barhop");
  });
  xit("Purpose links properly", () => {
    // this only works for buttons, not the expo-router link
    // will find a testing pattern for router based testing
    // render(<App />);
    // fireEvent.press(screen.getByText("Purpose"));
  });
});
