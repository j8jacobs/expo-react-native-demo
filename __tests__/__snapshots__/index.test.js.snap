// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "paddingVertical": 20,
      }
    }
  >
    <View>
      <Text
        href="/purpose"
        onPress={[Function]}
        role="link"
        style={
          {
            "borderRadius": 8,
            "borderWidth": 1,
            "height": 200,
            "margin": 12,
            "padding": 12,
            "width": "90%",
          }
        }
      >
        Purpose
      </Text>
      <Text
        href="/navigation"
        onPress={[Function]}
        role="link"
        style={
          {
            "borderRadius": 8,
            "borderWidth": 1,
            "height": 200,
            "margin": 12,
            "padding": 12,
            "width": "90%",
          }
        }
      >
        Navigation
      </Text>
      <Text
        href="/ye"
        onPress={[Function]}
        role="link"
        style={
          {
            "borderRadius": 8,
            "borderWidth": 1,
            "height": 200,
            "margin": 12,
            "padding": 12,
            "width": "90%",
          }
        }
      >
        Kanye West Quote Generator
      </Text>
      <Text
        href="/barhop"
        onPress={[Function]}
        role="link"
        style={
          {
            "borderRadius": 8,
            "borderWidth": 1,
            "height": 200,
            "margin": 12,
            "padding": 12,
            "width": "90%",
          }
        }
      >
        Build a Barhop
      </Text>
      <Text
        href="/notes/manage"
        onPress={[Function]}
        role="link"
        style={
          {
            "borderRadius": 8,
            "borderWidth": 1,
            "height": 200,
            "margin": 12,
            "padding": 12,
            "width": "90%",
          }
        }
      >
        Manage Notes
      </Text>
      <Text
        href="/animations"
        onPress={[Function]}
        role="link"
        style={
          {
            "borderRadius": 8,
            "borderWidth": 1,
            "height": 200,
            "margin": 12,
            "padding": 12,
            "width": "90%",
          }
        }
      >
        Animations
      </Text>
    </View>
  </RCTScrollView>
</View>
`;
